### PR TITLE
media-libs/lib3mf 2.1.0: version bump

### PR DESCRIPTION
**Pull request by waebbl**
Closes: https://bugs.gentoo.org/769275
Package-Manager: Portage-3.0.14, Repoman-3.0.2
Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

Current versions of media-gfx/openscad don't work with this API version and need to be restricted to <lib3mf-2. The upcoming openscad version bump does work with it. See
Bug: https://bugs.gentoo.org/769278

### CODE REVIEW

**Comment by waebbl** [comment]
Switching openscad live ebuild to use cmake instead of qmake. Works with both, lib3mf-1.8.1 and lib3mf-2.1.0. Latter needs a patch due to a case change from lib3MF.pc to lib3mf.pc.

**Comment by waebbl** [comment]
For the act dependency needed to re-add keywords for x86 and arm see https://github.com/gentoo/gentoo/pull/19411

**Comment by juippis** [comment]
Is the act dependency still needed? Can we somehow combine this PR with #19714?

**Comment by juippis** [comment]
Actually could you combine all the related work to a single PR. I'm at loss, and am fearing there'll be tree breakage or unnecessary rebuilds for users. Openscad is not a little package.

**Comment by waebbl** [comment]
@juippis Sure, I can combine them in a single PR.

The act package is 'only' needed to get x86 and arm keywords back to lib3mf. For amd64 the included binary does the job. Should I add it to the PR too?

**Comment by juippis** [comment]
Up to you, I quickly glanced at the ebuild and it might need .. a lot of fixes before it can be pushed. So maybe best to leave it out for now.

**Comment by waebbl** [comment]
Yes it might nieed some rework. The point with the package is, that upstream doesn't follow the usual way Go packages are developed, so the Go eclasses don't seem to be useful. On the other hand the build is rather trivial, so I decided for the way it's done.
I agree, it's better to leave it out for now and wait for a dedicated review of the package.

**Comment by juippis** [comment]
I've nearly caught up with our PR work queue, please ping me in ~a week from now for the act package and I'll take a look.

File: media-gfx/openscad/openscad-2019.05-r4.ebuild
Comment (juippis) [inline media-gfx/openscad/openscad-2019.05-r4.ebuild:49]: Will the update cause runtime issues? Ie should this be revbumped for this...

File: media-gfx/openscad/openscad-2021.01.ebuild
Comment (juippis) [inline media-gfx/openscad/openscad-2021.01.ebuild:46]: Does this work with both 1 and 2? Is some restriction needed? That's also why I asked to combine the PRs so that no modification is needed for this version after merging.

**Review by juippis — COMMENTED** [review]
Let me know the answers to my questions, there's no need for you to do anything, I can fix those minor issues while merging.

File: media-gfx/openscad/openscad-2019.05-r4.ebuild
Comment (waebbl) [reply media-gfx/openscad/openscad-2019.05-r4.ebuild:49]: 2019.05 will not build with lib3mf-2. Lib3mf-2 has been released after 2019.05 and support for it has not been backported

File: media-gfx/openscad/openscad-2021.01.ebuild
Comment (waebbl) [reply media-gfx/openscad/openscad-2021.01.ebuild:46]: The new version works with both lib3mf versions.

File: media-gfx/openscad/openscad-2019.05-r4.ebuild
Comment (juippis) [reply media-gfx/openscad/openscad-2019.05-r4.ebuild:49]: The idea is, people who have 2019.05-r4 installed now will update to lib3mf-2, because the restriction won't hit their package database before a rebuild. So, if lib3mf-2 will cause **runtime** issues to openscad-2019.05, this ebuilds needs t obe revbumped so that won't happen.

Comment (waebbl) [reply media-gfx/openscad/openscad-2019.05-r4.ebuild:49]: As the package won't build with lib3mf-2 I suspect it will cause runtime issues, whenever someone tries to im-/export a 3mf format file.
I'm gonna revbump the package then.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**
